pkg/security/security_profile/profile/manager.go

```diff
@@ -681,50 +683,76 @@ func (m *SecurityProfileManager) tryAutolearn(profile *SecurityProfile, event *m
 	var nodeType activity_tree.NodeGenerationType
 	var profileState EventFilteringProfileState
 
+	profile.eventTypeStateLock.Lock()
+	defer profile.eventTypeStateLock.Unlock()
+	eventState, ok := profile.eventTypeState[event.GetEventType()]
+	if !ok {
+		eventState = &EventTypeState{
+			lastAnomalyNano: profile.loadedNano,
+			state:           NoProfile,
+		}
+		profile.eventTypeState[event.GetEventType()] = eventState
+	} else if eventState.state == UnstableEventType {
+		// If for the given event type we already are on UnstableEventType, just return
+		// (once reached, this state is immutable)
+		m.incrementEventFilteringStat(event.GetEventType(), UnstableEventType, NA)
+		return UnstableEventType
+	}
+
 	// check if we are at the beginning of a workload lifetime
 	if event.ResolveEventTime().Sub(time.Unix(0, int64(event.ContainerContext.CreatedAt))) < m.config.RuntimeSecurity.AnomalyDetectionWorkloadWarmupPeriod {
 		nodeType = activity_tree.WorkloadWarmup
 		profileState = WorkloadWarmup
 	} else {
-		// have we reached the stable state time limit ?
-		lastAnomalyNano, ok := profile.lastAnomalyNano[event.GetEventType()]
-		if !ok {
-			profile.lastAnomalyNano[event.GetEventType()] = profile.loadedNano
-			lastAnomalyNano = profile.loadedNano
+		// If for the given event type we already are on StableEventType (and outside of the warmup period), just return
+		if eventState.state == StableEventType {
+			m.incrementEventFilteringStat(event.GetEventType(), StableEventType, NA)
```

Comment:
When we answer `StableEventType`, we already count the event later. I think we should remove this.

Reply:
thanks